File: Packages/RideSharingFeature/Sources/RideSharingFeature/DesignSystem/LiiveFareRow.swift
import SwiftUI

public struct LiiveFareRow: View {
    let label: String
    let amount: String
    var muted: Bool
    var total: Bool

    public init(label: String, amount: String, muted: Bool = false, total: Bool = false) {
        self.label = label
        self.amount = amount
        self.muted = muted
        self.total = total
    }

    public var body: some View {
        HStack(alignment: .firstTextBaseline, spacing: LiiveFareRowLayout.contentGap) {
            Text(label)
                .font(total ? LiiveFont.headline : LiiveFont.subhead)
                .fontWeight(total ? .semibold : .regular)
                .foregroundColor(labelColor)
            Spacer()
            Text(amount)
                .font((total ? LiiveFont.headline : LiiveFont.subhead).monospacedDigit())
                .fontWeight(total ? .bold : .medium)
                .foregroundColor(LiiveColor.text)
        }
        .padding(.top, total ? LiiveFareRowLayout.totalTopPadding : LiiveFareRowLayout.rowVerticalPadding)
        .padding(.bottom, total ? LiiveFareRowLayout.totalBottomPadding : LiiveFareRowLayout.rowVerticalPadding)
    }

    private var labelColor: Color {
        if total { return LiiveColor.text }
        return muted ? LiiveColor.textTertiary : LiiveColor.textSecondary
    }
}

private enum LiiveFareRowLayout {
    static let contentGap = LiiveSpacing.m
    static let rowVerticalPadding = LiiveSpacing.s - LiiveSpacing.xs2
    static let totalTopPadding = LiiveSpacing.m
    static let totalBottomPadding = CGFloat.zero
}

File: android/app/src/main/java/com/liive/ride/designsystem/LiiveFareRow.kt
package com.liive.ride.designsystem

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight

@Composable
fun LiiveFareRow(label: String, amount: String, muted: Boolean = false, total: Boolean = false) {
    val c = LiiveTheme.colors
    Row(
        Modifier
            .fillMaxWidth()
            .padding(
                top = if (total) LiiveFareRowLayout.TotalTopPadding else LiiveFareRowLayout.RowVerticalPadding,
                bottom = if (total) LiiveFareRowLayout.TotalBottomPadding else LiiveFareRowLayout.RowVerticalPadding
            ),
        horizontalArrangement = Arrangement.spacedBy(LiiveFareRowLayout.ContentGap)
    ) {
        Text(
            label,
            color = when {
                total -> c.text
                muted -> c.textTertiary
                else -> c.textSecondary
            },
            style = if (total) MaterialTheme.typography.titleLarge else MaterialTheme.typography.titleMedium,
            fontWeight = if (total) FontWeight.SemiBold else FontWeight.Normal,
            modifier = Modifier.alignByBaseline()
        )
        Spacer(Modifier.weight(1f))
        Text(
            amount,
            color = c.text,
            style = (if (total) MaterialTheme.typography.titleLarge else MaterialTheme.typography.titleMedium).tabularNumbers(),
            fontWeight = if (total) FontWeight.Bold else FontWeight.Medium,
            modifier = Modifier.alignByBaseline()
        )
    }
}

private object LiiveFareRowLayout {
    val ContentGap = LiiveSpacing.m
    val RowVerticalPadding = LiiveSpacing.s - LiiveSpacing.xs2
    val TotalTopPadding = LiiveSpacing.m
    val TotalBottomPadding = LiiveSpacing.xs2 - LiiveSpacing.xs2
}
